alerts/api.go

```diff
@@ -42,14 +42,25 @@ func NewTimeSpanFilter(start, stop time.Time) Filter {
 
 // NewInefficientResourceIDFilter provides a filter that matches on resource id.
 func NewInefficientResourceIDFilter(resourceID string) Filter {
-	return &filter{filterType: InefficientResourceIDFilter, value: resourceID}
+	return &filter{filterType: MatchResourceIDFilter, value: resourceID}
 }
 
 // NewCountSpanFilter provides a filter that matches on alert count.
 func NewCountSpanFilter(minCount, maxCount int64) Filter {
 	return &filter{filterType: CountSpanFilter, value: []int64{minCount, maxCount}}
 }
 
+// NewMinSeverityFilter provides a filter that matches on alert when severity is greater than
+// or equal to the minSev value.
+func NewMinSeverityFilter(minSev api.SeverityType) Filter {
```

Comment:
lets reduce the scope of minSeverityFilter to only specific resource types.
In this way we can only query a subtree.
For ex. All WARN node resource alerts

Reply:
i added a way to tell efficient filters how to filter after fetching. This is done using options and each of the efficient filters can take a number of options. So for instance, you could fetch using resource type and then have the fetched filters go through timeSpan check or severity check or both. Options provide a generic way to express such fetching/filtering for any of the efficient filters.